core/java/iesi-core/src/main/java/io/metadew/iesi/connection/database/netezza/NetezzaDatabaseService.java

```diff
@@ -18,16 +30,41 @@ public synchronized static NetezzaDatabaseService getInstance() {
         return INSTANCE;
     }
 
-    private NetezzaDatabaseService() {}
-
     @Override
     public NetezzaDatabase getDatabase(Connection connection) {
-        return null;
+        String userName = DatabaseHandler.getInstance().getMandatoryParameterWithKey(connection, USER_KEY);
+        String userPassword = DatabaseHandler.getInstance().getMandatoryParameterWithKey(connection, PASSWORD_KEY);
+        Optional<String> schemaName = DatabaseHandler.getInstance().getOptionalParameterWithKey(connection, SCHEMA_KEY);
```

Comment:
shouldn't this be database? shouldn't this be mandatory?

Reply:
https://www.ibm.com/docs/en/psfa/7.2.1?topic=SSULQD_7.2.1/com.ibm.nz.datacon.doc/r_datacon_jdbc_connection_string.html

Reply:
At this moment we are creating JDBC connection via ConnectionURL, Hence we don't need database here. Here Schema is optional because this is optional field in Netezza URL connection because Schema and database is different in Netezza database.
And on line 51 we are getting Database as mandatory parameter.